### PR TITLE
Jenkinsfile.release: support ARCHES parameter

### PR DESCRIPTION
**Pull request by dustymabe**
Add support for an ARCHES parameter that is a space spearated string
of architectures for us to target for the given build ID.

Add a loop over the architecture list where it makes sense.

### CODE REVIEW

File: Jenkinsfile.release [hunk 1]
```diff
@@ -74,85 +78,95 @@ podTemplate(cloud: 'openshift', label: pod_label, yaml: pod) {
...
-                    --family fedora-coreos-${params.STREAM} \
-                    --image "${gcp_image}"
-                """)
+            // For production streams, promote the GCP image so that it
+            // will be the chosen image in an image family and deprecate
+            // all others. `ore gcloud promote-image` does this for us.
+            if ((basearch == 'x86_64') && params.STREAM in streams.production) {
+                stage('GCP ${basearch}: Image Promotion') {
+                    if (meta.gcp?.image) {
+                        gcp_image = meta.gcp.image
```
Comment (jlebon) [inline Jenkinsfile.release:114]: Minor/optional: and also feels like we can just nuke that variable and use `meta.gcp.image` directly in the invocation?

File: Jenkinsfile.release [hunk 2]
```diff
@@ -74,85 +78,95 @@ podTemplate(cloud: 'openshift', label: pod_label, yaml: pod) {
...
-                cosa buildupload --build=${params.VERSION} --skip-builds-json s3 --acl=public-read ${s3_stream_dir}/builds
-                """)
+
+            if ((basearch in ['x86_64', 'aarch64']) && params.AWS_REPLICATION == 'true') {
+                // Replicate the newly uploaded AMI to other regions. Intentionally
+                // split out from the 'Upload AWS' stage to allow for tests to be added
+                // at a later date before replicating said image.
+                //
+                // We have to re-run the coreos-meta-translator as aws-replicate
+                // only modifies the meta.json
```
Comment (jlebon) [inline Jenkinsfile.release:136]: Minor: should move that comment down.

File: Jenkinsfile.release [hunk 3]
```diff
@@ -74,85 +78,95 @@ podTemplate(cloud: 'openshift', label: pod_label, yaml: pod) {
...
-          plume release --distro fcos \
-              --version ${params.VERSION} \
-              --stream ${params.STREAM} \
-              --bucket ${s3_bucket}
-          """)
+            // generate the release meta data and upload it back to s3
+            shwrap("""
+            cosa generate-release-meta --build-id ${params.VERSION} --workdir .
+            cosa buildupload --build=${params.VERSION} --skip-builds-json s3 --acl=public-read ${s3_stream_dir}/builds
+            """)
```
Comment (jlebon) [inline Jenkinsfile.release:151]: Minor: space?

File: Jenkinsfile.release [hunk 4]
```diff
@@ -74,85 +78,95 @@ podTemplate(cloud: 'openshift', label: pod_label, yaml: pod) {
...
+            // Run plume to publish official builds; This will handle modifying
+            // object ACLs, modifying AMI image attributes,
+            // and creating/modifying the releases.json metadata index
+            shwrap("""
+            export AWS_CONFIG_FILE=\${AWS_FCOS_BUILDS_BOT_CONFIG}
+            plume release --distro fcos \
+                --version ${params.VERSION} \
+                --stream ${params.STREAM} \
+                --bucket ${s3_bucket}
+            """)
```
Comment (jlebon) [inline Jenkinsfile.release:161]: Minor: space?

**Review by jlebon — APPROVED** [review]
LGTM overall, mostly minor nits.

File: Jenkinsfile.release [hunk 5]
```diff
@@ -74,85 +78,95 @@ podTemplate(cloud: 'openshift', label: pod_label, yaml: pod) {
...
         """)
 
-        // Fetch metadata for the build we are interested in
+        // Fetch metadata files for the build we are interested in
         stage('Fetch Metadata') {
             shwrap("""
             export AWS_CONFIG_FILE=\${AWS_FCOS_BUILDS_BOT_CONFIG}
             cosa init --branch ${params.STREAM} https://github.com/coreos/fedora-coreos-config
-            cosa buildprep --build=${params.VERSION} s3://${s3_stream_dir}/builds
+            cmd-buildprep --build=${params.VERSION} \
```
[the remark this answers is not in the document]
Comment (dustymabe) [reply Jenkinsfile.release:86]: ha - complete mistake will fix

File: Jenkinsfile.release (hunk 2 above)
Comment (dustymabe) [reply Jenkinsfile.release:136]: hmm - I don't see where we run coreos-meta-translator at all - old comment?

File: Jenkinsfile.release (hunk 3 above)
Comment (dustymabe) [reply Jenkinsfile.release:151]: not following.. you want a blank line to follow this code block?

File: Jenkinsfile.release (hunk 4 above)
Comment (dustymabe) [reply Jenkinsfile.release:161]: not following.. you want a blank line to follow this code block?

File: Jenkinsfile.release (hunk 1 above)
Comment (dustymabe) [reply Jenkinsfile.release:114]: ok tried to modify it to be better

File: Jenkinsfile.release [hunk 6]
```diff
@@ -74,85 +78,95 @@ podTemplate(cloud: 'openshift', label: pod_label, yaml: pod) {
...
-                    --json-key \${GCP_IMAGE_UPLOAD_CONFIG} \
-                    --family fedora-coreos-${params.STREAM} \
-                    --image "${gcp_image}"
-                """)
+            // For production streams, promote the GCP image so that it
+            // will be the chosen image in an image family and deprecate
+            // all others. `ore gcloud promote-image` does this for us.
+            if ((basearch == 'x86_64') && params.STREAM in streams.production) {
+                stage('GCP ${basearch}: Image Promotion') {
+                    if (meta.gcp?.image) {
```
[the remark this answers is not in the document]
Comment (dustymabe) [reply Jenkinsfile.release:113]: changed the if conditional and we shouldn't generate an empty stage

File: Jenkinsfile.release (hunk 2 above)
Comment (jlebon) [reply Jenkinsfile.release:134]: Yeah, old name for `cosa generate-release-meta`. I think the comment is still valuable. I meant moving it down to where we run `cosa generate-release-meta` now (and I guess update the comment too).

**Review by jlebon — APPROVED** [review]

File: Jenkinsfile.release (hunk 2 above)
Comment (dustymabe) [reply Jenkinsfile.release]: +1 added an updated comment below

**Review by jlebon — APPROVED** [review]

**Pull request merged**